### DISCUSSION TITLE
[CollectionView] 데이터 설정 질문입니다,,

### DISCUSSION THREAD

**Discussion by jinmoon23** [SwiftBreakers/TTEOPPOKKI4U#42] [Q&A]
블로그 2개 참조해서 보여지는 것 까지는 성공했는데 블로그의 코드 방식이 낯설어서인지 데이터 주입이 어렵네요.. 이것저것 하느라 그런지 머리가 잘 안돌아갑니다 흑흑
참고 블로그 : https://ios-development.tistory.com/974, https://ios-development.tistory.com/1021

1. firebase storage -> Recommend 이미지 저장
2. 새로운 도큐먼트를 생성한 후 collectionImageURL1~4 필드를 설정해 거기다가 1의 이미지를 넣을 예정이었습니다.

jinmoon 브랜치 푸시 완료하였습니다. 코드 참조해주세요. PS, 기능 구현에만 집중하다 보니 코드가 가독성이... 미리 죄송합니다. 
추가적으로 동영상도 첨부하였습니다. 

https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/161567425/2bc25df4-5965-4e20-87cd-ff9da6fca52a

**Comment by Haroldfromk** [comment]
저기 아래에는 어떤 이미지가 들어가는걸까요?? recommendMain이라는 Collection의 각각의 Documents에 새로운 ImageURL 배열이 값을 담는 Field가 추가되는걸까요??

만약 그렇게 새로운 ImageURL Field가 추가가 된다면 KingFisherManager를 통해 URL을 UIImage로 전환이 가능한데, 해당 메서드를 통해서 이미지로 바꾸면 될것같아요 관련 글은 https://terry-some.tistory.com/m/89 여기를 참고하시면 좋을것같네요 :)

실제로 현재 프로젝트 내에서 해당 라이브러리를 통해 URL to UIImage로 Conversion을 하는 사례가 있습니다. 프로젝트 내에서 KingFisherManager를 검색하시면 사용한 예시를 볼 수 있습니다 :)

![CleanShot 2024-06-15 at 16 42 28@2x](https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/97341336/79229cab-869f-42c2-b5c2-b72131514326)

물론 위방법은 URL을 이미지로 변환해서 추가하는것인데, 그게아니고 CollectionView의 Cell에 Imageview가 있다고 가정을 한다면
imageview.kf.setimage 메서드를 사용하면 될걸로 보입니다.

![CleanShot 2024-06-15 at 16 45 59@2x](https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/97341336/4dcd5dff-3074-4a58-9f4f-10e526a6d300)

두가지 예시로 필요한 방식에 맞게 구현을 하면 될거같아요

**Reply by jinmoon23** [reply]
감사합니다! 참고하겠습니다. 추가질문 폭격을 기대해주세요 ^_^

**Reply by Haroldfromk** [reply]
넵 :)
